lib/redis_memo/memoizable/dependency.rb

```diff
@@ -40,6 +43,20 @@ def depends_on(dependency, **conditions)
     end
   end
 
+  def extract_dependencies_for_relation(relation)
+    connection = ActiveRecord::Base.connection
+    query, binds = connection.send(:to_sql_and_binds, relation.arel) # Is there another way to get this info?
```

Comment:
I think this is the right method to use here! we need to thoroughly test the rails versions we support before remove the "beta" tag anyway 😄 

Note this method returns a tuple of 3: https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L42

sql, binds, preparable